api/src/main/java/com/velocitypowered/api/proxy/Player.java

```diff
@@ -45,6 +46,20 @@
    */
   String getUsername();
 
+  /**
+   * The locale the velocity proxy is translating its messages to. By default {@link PlayerSettings#getLocale()}
+   *
+   * <p>Can be null when the settings have not been initialized yet.</p>
+   *
+   * @return the locale.
+   *
+   */
+  @Nullable Locale getProxyLocale();
```

Comment:
I don't like this name - perhaps `getEffectiveLocale()`?

Reply:
Sounds like a good name. Changed.